server/channels/app/platform/log_test.go

```diff
@@ -36,6 +34,14 @@ func TestGetMattermostLog(t *testing.T) {
 	dir, err := os.MkdirTemp("", "")
 	require.NoError(t, err)
 	t.Cleanup(func() {
+		// MM-62438: Disable file target before cleaning up
+		// to avoid a race between removing the directory and the file
+		// getting written again.
+		th.Service.UpdateConfig(func(cfg *model.Config) {
+			*cfg.LogSettings.EnableFile = false
+		})
+		th.Service.Logger().Flush()
+
 		err = os.RemoveAll(dir)
```

Comment:
I wonder if `TestHelper.TearDown` should be the one clearing up the log files. It's a bit odd that this test is the one doing that.

Reply:
@hanzei The test is specifically testing a new temporary directory target for the logs so we create and remove it from within the test.